File: src/models/fee.py
class Fee:
    def __init__(self, name: str, amount : float, currency : str):
        self.name = name
        self.amount = amount
        self.currency = currency
        
    def __str__(self):
        return f"Fee(Name: {self.name}, Fee: {self.amount}, Currency: {self.currency})"
    
    def __repr__(self):
        return f"Fee(Name: {self.name}, amount: {self.amount}, Currency: {self.currency})"
    
    def to_dict(self):
        return {"name": self.name, "amount": self.amount, "currency": self.currency}
    
    def __eq__(self, other):
        return self.name == other.name and self.amount == other.amount and self.currency == other.currency
    
    def __hash__(self):
        return hash((self.name, self.amount, self.currency))

File: src/models/fee.ts
class Fee {
    name: string;
    amount: number;
    currency: string;
    exchangeRate? : number;
    constructor(name : string, amount : number, currency : string, exchangeRate? : number) {
        this.name = name;
        this.amount = amount;
        this.currency = currency;
        if(exchangeRate && !isNaN(exchangeRate)) this.exchangeRate = exchangeRate;
    }

    toString() {
        return `Fee(Name: ${this.name}, Amount: ${this.amount}, Currency: ${this.currency})`;
    }

    equals(other : Fee) : boolean {
        return this.name === other.name && this.amount === other.amount && this.currency === other.currency && this.exchangeRate === other.exchangeRate;
    }

    static isEquals(fee1 : Fee, fee2 : Fee) : boolean { // static isEquals(fee1, fee2) {
        return fee1.name === fee2.name && fee1.amount === fee2.amount && fee1.currency === fee2.currency && fee1.exchangeRate === fee2.exchangeRate;
    }
}

export { Fee };